scripts/irods/test/test_all_rules.py

```diff
@@ -1818,6 +1819,41 @@ def test_msiSetKeyValuePairsToObj_does_not_crash__issue_7027(self):
 
         self.admin.assert_icommand(['imeta', 'ls', '-d', TEST_FILE], 'STDOUT', [ 'attribute: key\nvalue: val\n' ])
 
+    @unittest.skipUnless(plugin_name == 'irods_rule_engine_plugin-python', "Requires the PREP.")
+    def test_msiDataObjChksum_does_not_lead_to_segfault_on_good_verification_result__issue_7859(self):
+        with temporary_core_file() as core:
+            attr_key = 'issue_7859'
+            expected_attr_units = 'issue_7859 is working'
+
+            # Add a rule that attempts to verify the checksum using msiDataObjChksum.
+            # If verification finds no issues, the checksum output variable will hold an empty string.
+            # The check for the empty string and attachment of metadata give the test a way to determine
+            # the outcome of the rule.
+            core.add_rule(textwrap.dedent(f'''
+            def pep_api_data_obj_trim_pre(rule_args, callback, rei):
+                logical_path = str(rule_args[2].objPath)
+                result = callback.msiDataObjChksum(logical_path, 'verifyChksum=', '')
+                if result['arguments'][2] == '':
+                    callback.msiModAVUMetadata('-d', logical_path, 'set', '{attr_key}', '{attr_key}', '{expected_attr_units}')
+            '''))
+
+            # Create a new data object.
+            data_object = 'issue_7859_data_object'
+            self.user0.assert_icommand(['itouch', data_object])
```

Comment:
Do we need to clean up this data object or the metadata at the end?

Reply:
No, it's handled for us because the data object is automatically placed in the session collection.